### ISSUE TITLE
--skip-path regexp is sometimes compared to the absolute path

### ISSUE THREAD

**Issue by izeau** [bridgecrewio/checkov#5231]
**Describe the issue**

According to the [documentation](https://www.checkov.io/2.Basics/CLI%20Command%20Reference.html), `--skip-path` refers to the “Path (file or directory) to skip, using regular expression logic, **relative to the current working directory**” (emphasis mine). That is indeed the case when using some runners (e.g. `dockerfile`), but not for others. For instance, the `terraform` runner uses an absolute path due to the root directory [being set using `os.path.abspath()`](https://github.com/bridgecrewio/checkov/blob/2df19723d80024544df918bf85cf73b88e5edec7/checkov/terraform/runner.py#L124).

**Examples**

<details>
<summary>setup</summary>

```bash
$ cd $(mktemp -d)
$ mkdir -p foo subdir
$ echo 'FROM foo' > subdir/Dockerfile
$ echo 'resource "aws_instance" "foo" {}' > subdir/main.tf
$ cp -r subdir foo
$ tree                    
.
├── foo
│   └── subdir
│       ├── Dockerfile
│       └── main.tf
└── subdir
    ├── Dockerfile
    └── main.tf

4 directories, 4 files
$ alias ckv="checkov --check CKV_AWS_79,CKV_DOCKER_2 --skip-download --quiet --compact --directory ."
$ ckv                                                                                     
terraform scan results:

Passed checks: 0, Failed checks: 2, Skipped checks: 0

Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /foo/subdir/main.tf:1-1
Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /subdir/main.tf:1-1
dockerfile scan results:

Passed checks: 0, Failed checks: 2, Skipped checks: 0

Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /subdir/Dockerfile.
        File: /subdir/Dockerfile:1-1
Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /foo/subdir/Dockerfile.
        File: /foo/subdir/Dockerfile:1-1
```

</details>

<details>
<summary><code>--skip-path subdir</code> skips everything</summary>

```bash
$ ckv --skip-path subdir 
```

</details>

<details>
<summary><code>--skip-path "^./subdir"</code> leaves duplicate terraform checks</summary>

```bash
$ ckv --skip-path "^./subdir"
terraform scan results:

Passed checks: 0, Failed checks: 2, Skipped checks: 0

Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /foo/subdir/main.tf:1-1
Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /subdir/main.tf:1-1
dockerfile scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /foo/subdir/Dockerfile.
        File: /foo/subdir/Dockerfile:1-1
```

</details>

<details>
<summary><code>--skip-path "${PWD}/subdir"</code> leaves duplicate dockerfile checks</summary>

```bash
$ ckv --skip-path "${PWD}/subdir"
terraform scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /foo/subdir/main.tf:1-1
dockerfile scan results:

Passed checks: 0, Failed checks: 2, Skipped checks: 0

Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /foo/subdir/Dockerfile.
        File: /foo/subdir/Dockerfile:1-1
Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /subdir/Dockerfile.
        File: /subdir/Dockerfile:1-1
```

</details>

<details>
<summary>getting proper results using both <code>--skip-path</code> arguments</summary>

```bash
ckv --skip-path "^./subdir" --skip-path "${PWD}/subdir"
terraform scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_AWS_79: "Ensure Instance Metadata Service Version 1 is not enabled"
        FAILED for resource: aws_instance.foo
        File: /foo/subdir/main.tf:1-1
dockerfile scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV_DOCKER_2: "Ensure that HEALTHCHECK instructions have been added to container images"
        FAILED for resource: /foo/subdir/Dockerfile.
        File: /foo/subdir/Dockerfile:1-1
```

</details>

TL;DR:

- `--skip-path subdir` skips `./foo/subdir` and `./subdir`.
- `--skip-path "^subdir"` skips the dockerfile checks in `./subdir`, but runs the terraform ones, leading to duplicates.
- `--skip-path "${PWD}/subdir"` skips the terraform checks in `./subdir`, but runs the dockerfile ones, leading to duplicates.

**Desktop (please complete the following information):**
 - OS: macOS 13.4 / Debian Buster
 - Checkov Version: 2.3.285

**Comment by izeau** [comment]
This is still happening on version macOS 14.1 / 3.1.38

**Comment by izeau** [comment]
This is still happening on macOS 14.3 and Checkov version 3.2.136
